mobile/src/components/course/course.tsx

```diff
@@ -121,47 +111,43 @@ const Course: FC = () => {
   }
 
   return (
-    <ScrollView>
-      <View style={styles.container}>
-        <Text style={styles.h1}>{course?.title}</Text>
-        <View style={styles.currentCategory}>
-          <CourseCategory
-            keyName={currentCategory?.key ?? 'unknown'}
-            name={currentCategory?.name ?? 'Unknown'}
-            isActive={false}
-          />
-          {hasEditCategoryPermission && (
-            <Pressable
-              style={styles.editIconContainer}
-              onPress={handleEditModeToggle}
-            >
-              <Icon width={25} height={25} name="edit" color="white" />
-            </Pressable>
-          )}
-        </View>
-
-        <Image
-          style={styles.image}
-          source={{
-            uri: course?.imageUrl ?? getImageUri(defaultCourseImage),
-          }}
-        />
-        <Text style={styles.h2}>About this course</Text>
-        {Boolean(course.description) && (
-          <CourseDescription
-            description={course.description}
-            isExpanded={isDescriptionExpanded}
-            handleExpandedToggle={handleExpandedToggle}
+    <View style={styles.container}>
+      <FlatList
+        ListHeaderComponent={(): ReactElement => (
+          <CourseHeader
             width={width}
+            course={course}
+            hasEditCategoryPermission={hasEditCategoryPermission}
+            onEditModeToggle={handleEditModeToggle}
           />
         )}
-        <CourseModules
-          courseModules={courseModules}
-          isLoading={moduleIsLoading}
-          onModulePress={handleModulePress}
-        />
-      </View>
-    </ScrollView>
+        data={courseModules}
+        keyExtractor={({ id }): string => id.toString()}
+        renderItem={({ item: module, index }): ReactElement => (
+          <Pressable
+            onPress={(): void => handleModulePress(module.courseId, module.id)}
+            disabled={!user}
+          >
+            <Module
+              index={index}
+              title={module.title}
+              description={module.description}
+            />
+          </Pressable>
+        )}
+        ListEmptyComponent={(): ReactElement =>
+          moduleIsLoading ? (
+            <Spinner />
```

Comment:
Maybe it would be better to move `<Spinner isOverflow />` higher, above FlatList component?

Reply:
We've already have one, while the course info is loading, this one is small only for modules loading, in case, that modules can load slowly, I think we don't have to hide all other course info undef the spinner

Reply:
Okey, got it